### ISSUE TITLE
deno lsp auto-importing does not work

### ISSUE THREAD

**Issue by luiskugel** [denoland/deno#26524]
Version: 2.0.2

I am using deno in a turborepo monorepo. I setup the deno lsp like the tuturial suggests and added some more settings:

![image](https://github.com/user-attachments/assets/3b92ee8e-febf-434a-96de-e931b7a964e6)

It works for normal code completion and also when i want to import "normal" node stuff. But when i want to code complete (and auto-import) for example this drizzle code, it just does not give me the option:

![image](https://github.com/user-attachments/assets/9371c795-adaf-4376-a787-2b41e7ec5859)

Even when there is already an import statement for drizzle. And it is not a problem with drizzle and its types. I tested it with a package from jsr and it also did not work. 

(The project is a former nodejs project. Do i need to do anything to set this up?)

**Comment by bartlomieju** [comment]
CC @nayeemrmn

**Comment by luiskugel** [comment]
For me it never worked. Not for the first import and as you can see also not when something (in this example the and function) is already imported. I also tried it without the "npm:" prefix and restarted the language server. 

If you can't reproduce it, i can try setting up a minimal demo.

**Comment by nayeemrmn** [comment]
It seems paths with `/node_modules/` in them are excluded from auto-import destinations by tsc, and we have to hack around this. Fix coming soon

**Comment by ctsstc** [comment]
I'm wondering if this has been fixed? I'm using Deno version: `2.4.4`

I have a basic `deno.json` like:
```json
{
  "imports": {
    "@hono/hono": "jsr:@hono/hono@^4.9.2",
  },
  "tasks": {
    "start": "deno run --allow-net main.ts"
  },
  "compilerOptions": {
    "jsx": "precompile",
    "jsxImportSource": "hono/jsx"
  }
}
```

Note: This project was created using the Hono generator with the Deno setup as a starting point.

When I go to type `const app = new Hono()` in a file I get no import hints it seems. 

- <img width="284" height="52" alt="Image" src="https://github.com/user-attachments/assets/6340b258-a085-473a-9c10-83ea4f66ef65" />
- <img width="427" height="166" alt="Image" src="https://github.com/user-attachments/assets/42b1364b-9bc8-4ede-97ca-b12a9d19994d" />

I also feel like imports are not showing up as I would expect them either for the auto complete suggestions.

<img width="321" height="127" alt="Image" src="https://github.com/user-attachments/assets/485d86ee-6e68-4211-9319-191fca4f546b" />

I've verified that I've enabled Deno for the project. Here's my `settings.json`.
```json
{
  "deno.enablePaths": [
    "./"
  ],
  "editor.inlayHints.enabled": "off",
  "deno.enable": true,
  "editor.defaultFormatter": "denoland.vscode-deno",
  "deno.codeLens.implementations": true,
  "deno.codeLens.references": true,
  "deno.codeLens.referencesAllFunctions": true,
  "deno.codeLens.test": true,
}
```

Here's my `deno.json`
```json
{
  "imports": {
    "@hono/hono": "jsr:@hono/hono@^4.9.2",
  },
  "tasks": {
    "start": "deno run --allow-net main.ts"
  },
  "compilerOptions": {
    "jsx": "precompile",
    "jsxImportSource": "hono/jsx"
  }
}
```

If I do finally get some hints after manually typing in the import from portion:

<img width="295" height="296" alt="Image" src="https://github.com/user-attachments/assets/bc571c7f-c60d-484c-9b9a-01d3473ee617" />

I also start getting proper autocomplete items within Hono after the import has been created, but none of these show up until that import line exists.

<img width="554" height="140" alt="Image" src="https://github.com/user-attachments/assets/1360a1c3-2dd3-4c69-af39-12cb69064dba" />

**Comment by nayeemrmn** [comment]
@ctsstc See #28775.
